src/lib/TopLevel.hs

```diff
@@ -909,7 +936,57 @@ getFilteredLogger = do
   logger  <- getLogger
   return $ FilteredLogger shouldLog logger
 
--- -- === instances ===
+-- === hoisting Ix dicts ===
+
+hoistIxDicts
+  :: ( Topper m, Mut n, GenericallyTraversableE e
+     , SinkableE e, SubstE Name e, HoistableE e)
+  => e n -> m n (e n)
+hoistIxDicts e = do
+  Abs frag (PairE e' (IxHoistingState UnitE)) <-
+    liftGenericTraverserMTopEmissions (IxHoistingState UnitE) $ traverseGenericE (sink e)
+  e'' <- emitEnv (Abs frag e')
+  evalRequiredSpecializations e''  -- track the needed sepcializations rather than traversing again to find them
```

Comment:
Is that comment a TODO, or am I missing something?

Reply:
Yep, you're right. It was meant to be a TODO